waspc/tools/migrate-0-11-to-0-12.js

```diff
@@ -0,0 +1,209 @@
+      const deps = waspDepsStr.match(/\(\s*"[^"]+"\s*,\s*"[^"]+"\s*\)/g).map((depStr) => {
+        const [packageNameStr, versionStr] = depStr.match(/"[^"]+"/g);
+        return {packageNameStr, versionStr};
+      });
+      const updatedMainWaspContent = mainWaspContent.replace(waspDepsStr, '');
+
+      const packageJsonDepsStr = deps.map(({ packageNameStr, versionStr }) => `    ${packageNameStr}: ${versionStr}`).join(',\n');
+      const packageJsonPath = path.join(projectDirName, 'package.json');
+      const packageJsonContent = fs.readFileSync(packageJsonPath, 'utf8');
+      const updatedPackageJsonContent = packageJsonContent.replace(/"dependencies"\s*:\s*\{/, `"dependencies": {\n${packageJsonDepsStr},`);
+
+      fs.writeFileSync(packageJsonPath, updatedPackageJsonContent);
+      fs.writeFileSync(mainWaspPath, updatedMainWaspContent);
+    }
+  }
+
+  {
+    console.log('11. Merging .gitignore from old project into .gitignore in new project...');
+    if (fs.existsSync(oldGitignorePath)) {
+      const oldGitignoreLines = fs.readFileSync(oldGitignorePath, 'utf8').split('\n');
+      const newGitignoreLines = fs.readFileSync(gitignorePath, 'utf8').split('\n');
+      const linesToAdd = oldGitignoreLines.filter((line) => !newGitignoreLines.includes(line));
+      fs.appendFileSync(gitignorePath, linesToAdd.join('\n'));
+
+    }
+  }
+
+  {
+    console.log('12. Copying the rest of relevant top level files and dirs...');
+    fs.readdirSync(oldProjectDirName).forEach((item) => {
+      if (!['.gitignore', 'main.wasp', 'src', '.wasp', '.waspinfo', '.wasproot', 'node_modules'].includes(item)) {
+        const oldItemPath = path.join(oldProjectDirName, item);
+        const newItemPath = path.join(projectDirName, item);
+        if (fs.lstatSync(oldItemPath).isDirectory()) {
+          fs.cpSync(oldItemPath, newItemPath, { recursive: true });
+        } else {
+          fs.copyFileSync(oldItemPath, newItemPath);
+        }
+      }
+    });
+  }
+
+  {
+    console.log(`13. Cleaning up (wasp clean, deleting ${oldProjectDirName} dir)...`);
+    execSync(`cd ${projectDirName} && ${WASP_BIN} clean`);
+    execSync(`rm -rf ${oldProjectDirName}`);
+  }
+
+  console.log(
+    '\nAll done! Please inspect the changes we did, ideally with `git diff`, to make sure they make sense.'
+    + '\nLook out for any TODO comments we might have generated for the JS imports, as those need to be resolved manually.'
+  );
+}
+
+function assertWaspVersion() {
+  if (!execSync(`${WASP_BIN} version`).toString().includes('0.12.')) {
+    console.error('Error: Your wasp is not version 0.12.x');
+    process.exit(1);
+  }
+}
+
+function obtainProjectDirName() {
+  const projectDirName = process.argv[2].replace(/\/$/, '');
+  if (projectDirName.split(path.sep).length > 1) {
+    console.error('Error: Please provide the name of the project, not the path.');
+    process.exit(1);
+  }
+  return projectDirName;
+}
+
+async function printWarning() {
+ console.log(
```

Comment:
NodeJS is a bit of PITA when you have to get user input via the command line, so I found this to be much simpler: do a countdown.